jsapp/js/account/plan.component.tsx

```diff
@@ -0,0 +1,440 @@
+import React, {useEffect, useReducer, useState} from 'react';
+import styles from './plan.module.scss';
+import {
+  getOrganization,
+  getProducts,
+  getSubscription,
+  Product,
+  Organization,
+  BaseSubscription,
+  postCheckout,
+  postCustomerPortal,
+  BasePrice,
+  Price,
+} from './stripe.api';
+import Icon from '../components/common/icon';
+
+interface PlanState {
+  isLoading: boolean;
+  subscribedProduct: BaseSubscription;
+  intervalFilter: string;
+  filterToggle: boolean;
+  products: Product[];
+  organization: null | Organization;
+}
+
+// An interface for our action
+interface dataUpdates {
+  type: string;
+  prodData?: any;
+}
+
+const initialState = {
+  isLoading: true,
+  subscribedProduct: [],
+  intervalFilter: 'year',
+  filterToggle: false,
+  products: [],
+  organization: null,
+};
+
+function planReducer(state: PlanState, action: dataUpdates) {
+  switch (action.type) {
+    case 'initialProd':
+      return {...state, products: action.prodData};
+    case 'initialOrg':
+      return {...state, organization: action.prodData};
+    case 'initialSub':
+      return {...state, subscribedProduct: action.prodData};
+    case 'month':
+      return {
+        ...state,
+        intervalFilter: 'month',
+        filterToggle: !state.filterToggle,
+      };
+    case 'year':
+      return {
+        ...state,
+        intervalFilter: 'year',
+        filterToggle: !state.filterToggle,
+      };
+    default:
+      return state;
+  }
+}
+
+export default function Plan() {
+  const [state, dispatch] = useReducer(planReducer, initialState);
+  const [expandComparison, setExpandComparison] = useState(false);
```

Comment:
I'm curious why you choose to call the top variable "state" which sounds like it might contain the entire component's state but then added more useState's. I'm not necessarily against this if you feel that these boolean states are very unrelated.

Reply:
So in a few examples for using useReducer it used "state" which I thought made sense. I added the other useStates instead of filtering it through the reducer because I felt that the reducer was being used for more data driven items while these were visual items